From e7379023caf0c7511327462b61fb9c85aa4f3881 Mon Sep 17 00:00:00 2001
From: zuoqian <zuoqian2032@phytium.com.cn>
Date: Thu, 27 Feb 2025 05:44:58 +0000
Subject: [PATCH 29/67] PHYTIUM: net/phytmac: Bugfix set WOL failed issue

Before configuring WOL, we need to obtain the packet types
that WOL supports.

Mainline: NA
Signed-off-by: Li Wencheng <liwencheng@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>

Signed-off-by: zuoqian <zuoqian2032@phytium.com.cn>
Link: https://gitee.com/phytium_embedded/phytium-linux-kernel/commit/88f00b76168d9ba6ee37c0286d5c750a4c6ab806
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/net/ethernet/phytium/phytmac_ethtool.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/phytium/phytmac_ethtool.c b/drivers/net/ethernet/phytium/phytmac_ethtool.c
index 8bba0af005be..e1698fa10b09 100644
--- a/drivers/net/ethernet/phytium/phytmac_ethtool.c
+++ b/drivers/net/ethernet/phytium/phytmac_ethtool.c
@@ -89,23 +89,23 @@ static void phytmac_get_wol(struct net_device *ndev, struct ethtool_wolinfo *wol
 {
 	struct phytmac *pdata = netdev_priv(ndev);

+	wol->wolopts = 0;
 	phylink_ethtool_get_wol(pdata->phylink, wol);

+	wol->supported = WAKE_MAGIC | WAKE_ARP |
+			 WAKE_UCAST | WAKE_MCAST;
+
 	if (pdata->wol & PHYTMAC_WAKE_MAGIC) {
 		wol->wolopts |= WAKE_MAGIC;
-		wol->supported |= WAKE_MAGIC;
 	}
 	if (pdata->wol & PHYTMAC_WAKE_ARP) {
 		wol->wolopts |= WAKE_ARP;
-		wol->supported |= WAKE_ARP;
 	}
 	if (pdata->wol & PHYTMAC_WAKE_UCAST) {
 		wol->wolopts |= WAKE_UCAST;
-		wol->supported |= WAKE_UCAST;
 	}
 	if (pdata->wol & PHYTMAC_WAKE_MCAST) {
 		wol->wolopts |= WAKE_MCAST;
-		wol->supported |= WAKE_MCAST;
 	}
 }

--
2.50.1
